### PR TITLE
#167893656 In-app and email notifications for reported articles

### PR DESCRIPTION
**Pull request by VictorMaria**
### What does this PR do?
Create in-app notification for the author of a reported article
Send email notification to the author of a reported article

#### Description of Task to be completed?
Has in-app and email notification functionality working
API Flowchart for reporting articles (2)

#### How should this be manually tested?
```
# Fetch this branch
$ git fetch origin ft-notifications-for-reported-articles-167893656

# Install dependencies
$ npm install

# Start server
$ npm run server
```
1. Launch two tabs on Postman, one to sign in and another to create an article.
2. Sign in on a third tab as a different user
3. Launch a fourth Postman tab and Paste /api/{slug}/report in the URL bar
4. Make a Post request with reportTitle and reportBody as the body of the request.
5. After a successful request you should receive an in-app and email notification

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167893656](https://www.pivotaltracker.com/story/show/167893656)

#### Screenshots (if appropriate)
Email notification
Screen Shot 2019-08-15 at 1 43 11 PM

In-app Notification
Screen Shot 2019-08-15 at 3 32 04 AM

### CODE REVIEW

**Review by PelumiAlesh — APPROVED** [review]
LGTM

File: controllers/ArticleController.js [hunk 1]
```diff
@@ -572,7 +572,16 @@ export default class ArticleController {
...
+          { // Author
+            model: User,
+            as: 'author',
+            attributes: ['id', 'email', 'firstName'],
+          }
+        ]
+      });
       if (!article) return errorResponse(res, 404, { message: 'Article not found' });
 
       const articleId = article.dataValues.id;
```
Comment (mekzy-o) [inline controllers/ArticleController.js:568]: Victor Try destructuring line 587

File: helpers/UrgentEmail.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+  *
+  * @static
+  * @param {string} recipient - recipient of the email
+  * @param {string} name - name of the recipient
+  * @param {object} payloadDetails - payload details
+  * @memberof EmailNotification
+  * @returns {function} - returns a function call
+  */
+  static sendUrgentEmail(recipient, name, payloadDetails) {
```
Comment (tolulope-od) [inline helpers/UrgentEmail.js]: Hi Victor, could this method name be changed to a more one? This one seems a little too vague

**Review by tolulope-od — COMMENTED** [review]
Good work Victor

File: helpers/UrgentEmail.js (hunk 2 above)
Comment (VictorMaria) [reply helpers/UrgentEmail.js]: I get your point, it has been changed. Thank you.

File: controllers/ArticleController.js (hunk 1 above)
Comment (VictorMaria) [reply controllers/ArticleController.js:568]: I will see to that. Thank you.

**Review by profmcdan — CHANGES_REQUESTED** [review]
Update your flowchart to reflect what was done and also edit the email template to properly format the date and the report details

**Review by mekzy-o — APPROVED** [review]
Nice Job Victor. LGTM

**Review by profmcdan — APPROVED** [review]

**Pull request merged**